### PR TITLE
Add state sync unit test

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR added unit test to compare two state/database.
If there is a difference, it will return error.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/state/sync_test.go [hunk 1]
```diff
@@ -39,7 +41,7 @@ type testAccount struct {
 }
 
 // makeTestState create a sample test state to test node-wise reconstruction.
-func makeTestState() (Database, common.Hash, []*testAccount) {
+func makeTestState(t *testing.T) (Database, common.Hash, []*testAccount) {
 	// Create an empty state
 	db := NewDatabase(database.NewMemoryDBManager())
 	statedb, _ := New(common.Hash{}, db)
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:47]:
```suggestion
	statedb, err := New(common.Hash{}, db)
	if err != nil {
		t.Fatlf(err.Error())
	}
```

File: blockchain/state/sync_test.go [hunk 2]
```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
 	return it.Error
 }
 
+// compareStatesConsistency checks that all data of given two states.
+func compareStatesConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash) error {
+	// Create and iterate a state trie rooted in a sub-node
+	if _, err := oldDB.ReadStateTrieNode(root.Bytes()); err != nil {
+		return err
+	}
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:173]: Are they necessary? The existence of root node could be checked with `New` function.

File: blockchain/state/sync_test.go [hunk 3]
```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
...
+	newState, err := New(root, NewDatabase(newDB))
+	if err != nil {
+		return err
+	}
+
+	oldIt := NewNodeIterator(oldState)
+	newIt := NewNodeIterator(newState)
+	for oldIt.Next() {
+		if !newIt.Next() {
+			return fmt.Errorf("newDB iterator finished early : oldIt.Hash(%v) oldIt.Parent(%v)", oldIt.Hash, oldIt.Parent)
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:191]:
```suggestion
			return fmt.Errorf("newDB iterator finished earlier : oldIt.Hash(%v) oldIt.Parent(%v)", oldIt.Hash, oldIt.Parent)
```

File: blockchain/state/sync_test.go [hunk 4]
```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
...
+			return fmt.Errorf("mismatched hash oldIt.Hash : oldIt.Hash(%v) newIt.Hash(%v)", oldIt.Hash, newIt.Hash)
+		}
+
+		if oldIt.Parent != newIt.Parent {
+			return fmt.Errorf("mismatched parent hash : oldIt.Parent(%v) newIt.Parent(%v)", oldIt.Parent, newIt.Parent)
+		}
+	}
+
+	if newIt.Next() {
+		return fmt.Errorf("oldDB iterator finished early  : newIt.Hash(%v) newIt.Parent(%v)", newIt.Hash, newIt.Parent)
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:204]:
```suggestion
		return fmt.Errorf("oldDB iterator finished earlier : newIt.Hash(%v) newIt.Parent(%v)", newIt.Hash, newIt.Parent)
```

File: blockchain/state/sync_test.go [hunk 5]
```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
...
+			return fmt.Errorf("newDB iterator finished early : oldIt.Hash(%v) oldIt.Parent(%v)", oldIt.Hash, oldIt.Parent)
+		}
+
+		if oldIt.Hash != newIt.Hash {
+			return fmt.Errorf("mismatched hash oldIt.Hash : oldIt.Hash(%v) newIt.Hash(%v)", oldIt.Hash, newIt.Hash)
+		}
+
+		if oldIt.Parent != newIt.Parent {
+			return fmt.Errorf("mismatched parent hash : oldIt.Parent(%v) newIt.Parent(%v)", oldIt.Parent, newIt.Parent)
+		}
```
Comment (aidan-kwon) [inline blockchain/state/sync_test.go:200]: Why don't you check other field values such as `code`?

File: blockchain/state/sync_test.go (hunk 2 above)
Comment (KimKyungup) [reply blockchain/state/sync_test.go:173]: Good. :) It is already checked in `oldState, err := New(root, NewDatabase(oldDB))`. 
I remove them.

File: blockchain/state/sync_test.go (hunk 5 above)
Comment (KimKyungup) [reply blockchain/state/sync_test.go:197]: Good point. I will add. :)

**Pull request merged**